config/always-sed.m4

```diff
@@ -0,0 +1,16 @@
+dnl #
+dnl # Set the flags used for sed in-place edits.
+dnl #
+AC_DEFUN([ZFS_AC_CONFIG_ALWAYS_SED], [
+	ac_inplace=""
+	AC_CACHE_CHECK([for sed --in-place], [ac_cv_path_SED],
+		[AC_PATH_PROGS_FEATURE_CHECK([SED], [sed],
+			[[tmpfile=$(mktemp)
+			  echo foo > $tmpfile
+			  $ac_path_SED --in-place 's#foo#bar#' $tmpfile \
+			  && ac_cv_path_SED=$ac_path_SED
+			  rm $tmpfile]],
+			[ac_inplace="-i ''"])])
+	AS_IF([test "x$ac_inplace" = "x"], [ac_inplace="--in-place"])
```

Comment:
I'm confused by this, so take this with a grain of salt, but can't we just use `sed -i` everywhere rather than `sed --in-place`?

Reply:
FreeBSD and GNU sed behave somewhat differently.

Reply:
I did some testing just now. `sed -i'' FILENAME` works on GNU sed, but `sed -i '' FILENAME` does not. If `sed -i''` works on FreeBSD, we could likely use that everywhere.

Reply:
it doesn't work, and it is a trivial issue, we should just move on.

Reply:
i don't know about `--in-place` but it doesn't even exist on freebsd sed, this is probably compatibility for some other userland.

Reply:
`--in-place` is what was there originally, and is a GNU sed option. This code is changing that to `-i ''` (note the space) for FreeBSD sed. In GNU sed, the argument to -i is optional, so it must be specified immediately adjacent to it, as `-i''`. My point is that if `-i''` works on FreeBSD sed (and it likely does, given the `-i.bak` earlier, then we can avoid this entire autoconf test and the replacements every time we need `sed -i`.

Reply:
yes, and it does not work, as I've tested on FreeBSD today.

Reply:
Right. I wasn't thinking this through properly. `sed -i''` is exactly the same as `sed -i`, because the shell handles the quoting. This makes sense. Thanks for testing!